staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5656,6 +5717,27 @@ type NodeDaemonEndpoints struct {
 	KubeletEndpoint DaemonEndpoint `json:"kubeletEndpoint,omitempty" protobuf:"bytes,1,opt,name=kubeletEndpoint"`
 }
 
+// RuntimeClassFeatures is a set of runtime features.
+type RuntimeClassFeatures struct {
+	// RecursiveReadOnlyMounts is set to true if the runtime class supports RecursiveReadOnlyMounts.
+	// +optional
+	RecursiveReadOnlyMounts *bool `json:"recursiveReadOnlyMounts,omitempty" protobuf:"varint,1,opt,name=recursiveReadOnlyMounts"`
+	// UserNamespaces is set to true if the runtime class supports UserNamespaces.
+	// +optional
+	UserNamespaces *bool `json:"userNamespaces,omitempty" protobuf:"varint,2,opt,name=userNamespaces"`
```

Comment:
(This field was not present in the original KEP, but added for consistency with the latest CRI API https://github.com/kubernetes/cri-api/commit/c3d03b1349992a948b39080898e4f9ed06429bbe)

Reply:
is it possible to have an OR on the feature gates? Or maybe we just leave the RuntimeClassFeatures object ungated and gate the individual fields?

Reply:
We can but there has to be one release where the type is supported, but not used by default.  It's hard to reason about, so we probably shouldn't.

Reply:
what would you suggest we do for this? we have an object whose two fields are only relevant to two different features

Reply:
Ideally we would have two different gates - one controlling the introduction of the RuntimeClass stuff, and the other on RRO.  Otherwise we just lump them together and you can't get RuntimeClassFeatures until RRO goes beta.  RRO seems like a low risk of delay and can maybe even jump straight to GA.

Reply:
2 gates mean 2 KEPs for PRR - so keep that in mind :)

Reply:
not NECESSARILY but usually yes.